Sources/SwifterSwift/UIKit/UIViewExtensions.swift

```diff
@@ -233,6 +233,53 @@ public extension UIView {
         }
     }
 
+    /// SwifterSwift: Return constraints for this view. This must enumerate superviews.
+    var allConstraints: [NSLayoutConstraint] {
+        // https://stackoverflow.com/questions/47053727
+
+        // build array of self and all superviews
+        var views = [self]
+        var current = self
+        while let superview = current.superview {
+            views.append(superview)
+            current = superview
+        }
+
+        // now collect constraints affecting this view
+        return views.flatMap { $0.constraints }.filter {
+            $0.firstItem as? UIView == self || $0.secondItem as? UIView == self
+        }
+    }
+
+    /// SwifterSwift: First width constraint for this view
+    var widthConstraint: NSLayoutConstraint? {
+        findConstraint(attribute: .width)
+    }
+
+    /// SwifterSwift: First height constraint for this view
+    var heightConstraint: NSLayoutConstraint? {
+        findConstraint(attribute: .height)
+    }
+
+    /// SwifterSwift: First leading constraint for this view
+    var leadingConstraint: NSLayoutConstraint? {
+        findConstraint(attribute: .leading)
+    }
+
+    /// SwifterSwift: First trailing constraint for this view
+    var trailingConstraint: NSLayoutConstraint? {
+        findConstraint(attribute: .trailing)
+    }
+
+    /// SwifterSwift: First top constraint for this view
+    var topConstraint: NSLayoutConstraint? {
+        findConstraint(attribute: .top)
+    }
+
+    /// SwifterSwift: First bottom constraint for this view
```

Comment:
We normally have a rules where all extensions should be self-contained, but for these cases, since they are all related I'm not sure if we should apply it, because then the code will become such a mess ...

Reply:
would it help to group these in an extension down below?

Reply:
Sure! Is better if related are all inn the same place :)

Reply:
this is done, thanks